ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -103,12 +121,12 @@ private static void registerAuthHandler(
     router.route().handler(routingContext);
   }
 
-  private static Handler<RoutingContext> wrappedHandler(
+  private static Handler<RoutingContext> selectiveHandler(
       final Handler<RoutingContext> handler,
-      final Provider provider
+      final Predicate<Provider> providerTest
   ) {
     return rc -> {
-      if (rc.data().get(PROVIDER_KEY) != provider) {
+      if (!providerTest.test((Provider)rc.data().get(PROVIDER_KEY))) {
```

Comment:
I know you didn't change this part of the logic, but this looks slightly scary that it skips this handler if the particular provider is different from what we expect.  It just seems a little error prone that we decide which auth to use and invoke a handler in two different places.  Maybe this is because we have to statically install all the handers, but only invoke one of them at request-time? Just a general comment -- not sure if you have any ideas.

Reply:
Yes, I totally agree. This could be a vertx-ish architectural style, but I'm not sure if it is achieving anything in our setting. It should be possible to have a single wrapping handler to call the sub-handlers -- essentially, merge `selectHandler` and `selectiveHandler` into one, so that you can use plain old control-flow for handler selection instead of passing around `PROVIDER` values. I didn't want to make this too large of a refactoring, but if you want to go down this road with me, I could try.

Reply:
No need for anything in this PR, more just thoughts while reading surrounding code.  We can consider that as a followup if you do more changes in this area.